### ISSUE TITLE
Non unique index should not appear in findOne where clause

### ISSUE THREAD

**Issue by Sytten** [prisma/prisma#3869]
## Bug description

The `findOne` `where` clause contains `@@index([column])` indexes. This is wrong since a normal index doesn't have the `unique` constraint and is thus not guaranteed to find only one element.

## How to reproduce
```prisma
model Calendar {
  id         String @id @default(cuid())
  providerId String
  @@index([providerId])
}
```

Generate the client and try to use `findOne`, you will see the `providerId`.

## Expected behavior

This column should only appear in the `findMany` and `findFirst`.

## Environment & setup

- Database: PostgreSQL
- Prisma version: 2.6.2

**Comment by mavilein** [comment]
This seems to be the same as https://github.com/prisma/prisma-client-js/issues/711
Therefore directly putting it into the current sprint so we can have a look at that.

**Comment by dpetrick** [comment]
Fix available from `2.9.0-dev.58` onward. Will be included in the 2.9.0 release.
